infrastructure/templates/datafactory/components/db-maintenance-template.json

```diff
@@ -0,0 +1,352 @@
+            "typeProperties": {
+              "storedProcedureName": "[[dbo].[RemoveSoftDeletedSubjects]"
+            },
+            "linkedServiceName": {
+              "referenceName": "ls_sql_statistics",
+              "type": "LinkedServiceReference"
+            }
+          }
+        ],
+        "annotations": []
+      },
+      "dependsOn": [
+        "[concat(variables('factoryId'), '/linkedServices/ls_sql_statistics')]"
+      ]
+    },
+    {
+      "name": "[concat(parameters('factoryName'), '/purge_soft_deleted_subjects_trigger')]",
+      "type": "Microsoft.DataFactory/factories/triggers",
+      "apiVersion": "2018-06-01",
+      "properties": {
+        "annotations": [],
+        "runtimeState": "Started",
+        "pipelines": [
+          {
+            "pipelineReference": {
+              "referenceName": "pl_purge_subjects_statistics",
+              "type": "PipelineReference"
+            },
+            "parameters": {}
+          },
+          {
+            "pipelineReference": {
+              "referenceName": "pl_purge_subjects_public_statistics",
+              "type": "PipelineReference"
+            },
+            "parameters": {}
+          }
+        ],
+        "type": "ScheduleTrigger",
+        "typeProperties": {
+          "recurrence": {
+            "frequency": "Week",
+            "interval": 1,
+            "startTime": "2020-09-19T23:00:00.000Z",
+            "timeZone": "UTC",
+            "schedule": {}
+          }
+        }
+      },
+      "dependsOn": [
+        "[concat(variables('factoryId'), '/pipelines/pl_purge_subjects_statistics')]",
+        "[concat(variables('factoryId'), '/pipelines/pl_purge_subjects_public_statistics')]"
+      ]
+    },
+    {
+      "name": "[concat(parameters('factoryName'), '/rebuild_indexes_trigger')]",
+      "type": "Microsoft.DataFactory/factories/triggers",
+      "apiVersion": "2018-06-01",
+      "properties": {
+        "annotations": [],
+        "runtimeState": "Started",
+        "pipelines": [
+          {
+            "pipelineReference": {
+              "referenceName": "pl_rebuild_statistics_indexes",
+              "type": "PipelineReference"
+            },
+            "parameters": {
+              "Percentage": "[variables('fragmentationPercentage')]",
+              "Tables": "[variables('fragmentationTables')]"
+            }
+          },
+          {
+            "pipelineReference": {
+              "referenceName": "pl_rebuild_public_statistics_indexes",
+              "type": "PipelineReference"
+            },
+            "parameters": {
+              "Percentage": "[variables('fragmentationPercentage')]",
+              "Tables": "[variables('fragmentationTables')]"
+            }
+          }
+        ],
+        "type": "ScheduleTrigger",
+        "typeProperties": {
+          "recurrence": {
+            "frequency": "Week",
+            "interval": 1,
+            "startTime": "2020-09-18T12:00:00.000Z",
```

Comment:
ok changed it to 23:00:00 UTC & also set timeout on the activities to 4 - see how we go - can always adjust those. I'm wondering if it will stop the stored proc if it times out - hopefully - can experiment with it once deployed.